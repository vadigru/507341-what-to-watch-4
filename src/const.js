export const MOVIES_DEFAULT_AMOUNT = 8;
export const MOVIES_STEP_AMOUNT = 8;
export const MAX_GENRES_AMOUNT = 9;
export const ALL_GENRES = `All genres`;
export const START_DELAY = 1000;

export const Tab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const DefaultTab = {
  GENRE: `All genres`,
  MOVIE: `Overview`
};

export const RatingLevel = {
  ZERO: 0,
  BAD: 3,
  NORMAL: 5,
  GOOD: 8,
  AWESOME: 10
};

export const AppRoute = {
  ROOT: `/`,
  MOVIE_PAGE: `/movie`,
  SIGN_IN: `/login`,
  ADD_REVIEW: `/review`,
};
