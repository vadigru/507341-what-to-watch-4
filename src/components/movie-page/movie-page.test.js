import React from "react";
import renderer from "react-test-renderer";
import MoviePage from "./movie-page.jsx";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {ALL_GENRES, MOVIES_DEFAULT_AMOUNT} from "../../const.js";
import Namespace from "../../reducer/namespace.js";

const mockStore = configureStore([]);

const films = [
  {
    title: `Movie Name`,
    posterUrl: `https://url.com`,
    backgroundUrl: `https://url.com`,
    backgroundColor: `some color`,
    previewUrl: `https://url.com`,
    previewImage: `https://url.com`,
    genre: `genre`,
    release: 2020,
    director: `Famous Director`,
    starring: [`Actor One`, `Actor Two`, `Actor Three`],
    time: `1h 30m`,
    rating: 10,
    votes: 1000000,
    description: `Some Description`,
    id: 1,
    isFavorite: true,
    videoUrl: `https://url.com`,
  },
  {
    title: `Movie Name`,
    posterUrl: `https://url.com`,
    backgroundUrl: `https://url.com`,
    backgroundColor: `some color`,
    previewUrl: `https://url.com`,
    previewImage: `https://url.com`,
    genre: `genre 1`,
    release: 2020,
    director: `Famous Director`,
    starring: [`Actor One`, `Actor Two`, `Actor Three`],
    time: `1h 30m`,
    rating: 10,
    votes: 1000000,
    description: `Some Description`,
    id: 1,
    isFavorite: true,
    videoUrl: `https://url.com`,
  },
  {
    title: `Movie Name`,
    posterUrl: `https://url.com`,
    backgroundUrl: `https://url.com`,
    backgroundColor: `some color`,
    previewUrl: `https://url.com`,
    previewImage: `https://url.com`,
    genre: `genre 2`,
    release: 2020,
    director: `Famous Director`,
    starring: [`Actor One`, `Actor Two`, `Actor Three`],
    time: `1h 30m`,
    rating: 10,
    votes: 1000000,
    description: `Some Description`,
    id: 1,
    isFavorite: true,
    videoUrl: `https://url.com`,
  },
  {
    title: `Movie Name`,
    posterUrl: `https://url.com`,
    backgroundUrl: `https://url.com`,
    backgroundColor: `some color`,
    previewUrl: `https://url.com`,
    previewImage: `https://url.com`,
    genre: `genre 3`,
    release: 2020,
    director: `Famous Director`,
    starring: [`Actor One`, `Actor Two`, `Actor Three`],
    time: `1h 30m`,
    rating: 10,
    votes: 1000000,
    description: `Some Description`,
    id: 1,
    isFavorite: true,
    videoUrl: `https://url.com`,
  },
];

const movie = {
  title: `Movie Name`,
  posterUrl: `https://url.com`,
  backgroundUrl: `https://url.com`,
  backgroundColor: `some color`,
  previewUrl: `https://url.com`,
  previewImage: `https://url.com`,
  genre: `genre`,
  release: 2020,
  director: `Famous Director`,
  starring: [`Actor One`, `Actor Two`, `Actor Three`],
  time: `1h 30m`,
  rating: 10,
  votes: 1000000,
  description: `Some Description`,
  id: 1,
  isFavorite: true,
  videoUrl: `https://url.com`,
};


it(`Should render MoviePage component`, () => {
  const store = mockStore({
    [Namespace.DATA]: {
      films,
      promo: films[0]
    },
    [Namespace.STATE]: {
      genre: ALL_GENRES,
      showedMovies: MOVIES_DEFAULT_AMOUNT
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <MoviePage
            movies={films}
            movie={movie}
            previewImage={movie.previewImage}
            onMovieCardClick={() => {}}
            activeTab={`Overview`}
            onTabClick={() => {}}
            isBigPlayerActive={false}
            onBigPlayerOnOff={() => {}}
          />
        </Provider>
    )
  .toJSON();

  expect(tree).toMatchSnapshot();
});
