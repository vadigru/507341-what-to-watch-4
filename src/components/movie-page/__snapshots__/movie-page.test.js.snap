// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MoviePage component 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "some color",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Movie Name"
          src="https://url.com"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link "
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            className="user-block__link"
            href="/login"
            onClick={[Function]}
          >
            Sign in
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Movie Name
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              genre
            </span>
            <span
              className="movie-card__year"
            >
              2020
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="14"
                viewBox="0 0 18 14"
                width="18"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Movie Name"
            height="327"
            src="https://url.com"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <ul
                className="movie-nav__list"
              >
                <li
                  className="movie-nav__item movie-nav__item--active"
                >
                  <a
                    className="movie-nav__link"
                    href="#"
                    onClick={[Function]}
                  >
                    Overview
                  </a>
                </li>
                <li
                  className="movie-nav__item false"
                >
                  <a
                    className="movie-nav__link"
                    href="#"
                    onClick={[Function]}
                  >
                    Details
                  </a>
                </li>
                <li
                  className="movie-nav__item false"
                >
                  <a
                    className="movie-nav__link"
                    href="#"
                    onClick={[Function]}
                  >
                    Reviews
                  </a>
                </li>
              </ul>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              10
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Awesome
              </span>
              <span
                className="movie-rating__count"
              >
                1000000
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              Some Description
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                Famous Director
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring:
                Actor One, Actor Two, Actor Three
                 and other
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
